File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/ProjectRequest.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import java.time.LocalDate;
import java.util.List;
import java.util.UUID; /**
 * @author Jôph Yamba
 */
public record ProjectRequest(
        UUID portfolioId,
        String title,
        String description,
        List<String> imageUrls,
        LocalDate startAt,
        LocalDate completedAt
) {}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/ProjectRequest.kt
package dev.jefy.connectpro.portfolio.application.dtos

import java.util.*

data class ProjectRequest(
    val portfolioId: UUID,
    val name: String,
    val description: String,
){
    init {
        require(name.isNotBlank()) { "title must not be blank" }
        require(description.isNotBlank()) { "title must not be blank" }
    }
}